src/mesido/esdl/asset_to_component_base.py

```diff
@@ -504,6 +512,30 @@ def _get_connected_q_nominal(self, asset: Asset) -> Union[float, Dict]:
                 raise _RetryLaterException(
                     f"Could not determine nominal discharge for {asset.asset_type} '{asset.name}'"
                 )
+        elif len(asset.in_ports) == 2 and len(asset.out_ports) == 1:  # for gas_boiler
+            q_nominals = {}
+            try:
+                for port in asset.in_ports:
+                    connected_port = asset.in_ports[0].connectedTo[0]
+                    if isinstance(port.carrier, esdl.GasCommodity):
+                        q_nominals["Q_nominal_gas"] = self._port_to_q_nominal[connected_port]
+                        self._port_to_q_nominal[port] = q_nominals["Q_nominal_gas"]
+                    elif isinstance(port.carrier, esdl.HeatCommodity):
+                        q_nominals["Q_nominal"] = self._port_to_q_nominal[connected_port]
+                        self._port_to_q_nominal[port] = q_nominals["Q_nominal"]
+                    else:
+                        logger.error(f"{asset.name} has wrong carrier type specified on in port")
```

Comment:
For the electric boiler you also get in this if statement right and not only for gas_boiler? Thus I think the error should not be an error, if there is atleast one port with gas or heat.

Reply:
agree, done